src/DomainHost.sol

```diff
@@ -336,18 +364,27 @@ abstract contract DomainHost {
         emit Cage();
     }
 
-    /// @notice Set this domain's cure value
+    /// @notice Process this domain debt
     /// @param _lid Local ordering id
-    /// @param value The value of the cure [RAD]
-    function _tell(uint256 _lid, uint256 value) internal ordered(_lid) {
+    /// @param debt The value of debt [WAD]
+    function _tell(uint256 _lid, uint256 debt) internal ordered(_lid) {
         require(live == 0, "DomainHost/live");
-        require(!cureReported, "DomainHost/cure-reported");
-        require(_divup(value, RAY) <= grain, "DomainHost/cure-bad-value");
+        require(!debtReported, "DomainHost/debt-reported");
+
+        int256 gap = _int256(grain) - _int256(debt);
+
+        if (gap > 0) {
+            dai.transferFrom(address(escrow), address(this), uint256(gap));
+            daiJoin.join(address(vow), uint256(gap));
+        } else if (gap < 0) {
+            vat.suck(vow, address(this), uint256(-gap) * RAY);
+            daiJoin.exit(address(escrow), uint256(-gap));
+        }
```

Comment:
I'm not too keen on this structure as this makes the same attack vector as permissionless `rectify()`. I realize this is tricky because it's possible we are in global settlement state due to governance attack and cannot rely on an authed person fixing this.

Reply:
Yeah I'm leaning on removing this as well too, that's why I haven't made a single test case for it yet. It would be the naive way to do it, but yeah super risky.